### ISSUE TITLE
__debug__ symbol not recognized

### ISSUE THREAD

**Issue by RianKojaASQ** [astral-sh/ty#577]
### Summary

Just got this warning on vscode:

`Name \`__debug__\` used when not definedty[unresolved-reference](https://ty.dev/rules#unresolved-reference)`

Recalling the docs:  https://docs.python.org/3/library/constants.html

**__debug__**
This constant is true if Python was not started with an [-O](https://docs.python.org/3/using/cmdline.html#cmdoption-O) option. See also the [assert](https://docs.python.org/3/reference/simple_stmts.html#assert) statement.

### Version

0.0.1a8

**Comment by AlexWaygood** [comment]
This should be easy to fix -- it's an implicit global symbol that doesn't appear in typeshed's stubs, similar to `__builtins__`. We need to add a branch here like the one for `__builtins__`: https://github.com/astral-sh/ruff/blob/e8ea40012afda872ef9bdf496a8b888ba80b533b/crates/ty_python_semantic/src/symbol.rs#L327-L328

**Comment by jelle-openai** [comment]
Note `__debug__` is a little more special than that:

```
>>> __debug__ = 1
  File "<stdin>", line 1
SyntaxError: cannot assign to __debug__
```

Up to you how important it is to replicate that special treatment in ty, though.

**Comment by AlexWaygood** [comment]
We already implement the syntax error! https://play.ty.dev/1e89682b-1ef8-464b-b7a6-7814cdd37237

**Comment by suneettipirneni** [comment]
I'd like to work on this issue if it's not already in progress.

**Comment by carljm** [comment]
@suneettipirneni It's yours! Feel free to ask here or [in our Discord server](https://discord.com/invite/astral-sh) if you have questions about the implementation.
